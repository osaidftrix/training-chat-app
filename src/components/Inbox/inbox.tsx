import useChat from '../../hooks/useChat';
import ChatWindow from './ChatWindow/chatWindow';
import UserList from './UserList/userlist';

const Inbox = () => {
  const { toggledUser, setToggledUser } = useChat();

  return (
    <div className="container mx-auto shadow-lg rounded-lg">
      <div className="flex flex-row justify-between">
        <UserList toggledUser={toggledUser} setToggledUser={setToggledUser} />
        <ChatWindow toggledUser={toggledUser} setToggledUser={setToggledUser} />
      </div>
    </div>
  );
};

export default Inbox;
