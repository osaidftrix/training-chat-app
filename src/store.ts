import { combineReducers, configureStore } from '@reduxjs/toolkit';
import connectionSlice from './store/slices/connectionSlice';
import storage from 'redux-persist/lib/storage';
//import storage from 'redux-persist-indexeddb-storage';
import { persistReducer, persistStore } from 'redux-persist';
import thunk from 'redux-thunk';
const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['connection'],
  // whitelist: ['connection']
};

const rootReducer = combineReducers({
  connection: connectionSlice,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: [thunk],
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;

export const persistor = persistStore(store);
