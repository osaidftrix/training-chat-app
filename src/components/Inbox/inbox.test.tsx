import { describe, test, expect } from 'vitest';
import Inbox from './inbox';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { PersistGate } from 'redux-persist/integration/react';
import { store, persistor } from '../../store';

describe('<Inbox />', () => {
  test('Inbox mounts properly', () => {
    const renderer = create(
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <Inbox />
        </PersistGate>
      </Provider>,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
