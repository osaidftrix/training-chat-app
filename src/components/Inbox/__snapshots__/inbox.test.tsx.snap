// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Inbox /> > Inbox mounts properly 1`] = `
<div
  className="container mx-auto shadow-lg rounded-lg"
>
  <div
    className="flex flex-row justify-between"
  >
    <div
      className="flex flex-col w-2/5 border-r-2 overflow-y-auto"
    >
      <div
        className="flex justify-end cursor-pointer"
      >
        <span
          className="mr-1"
        >
          Refresh Users
        </span>
        <svg
          className=""
          fill="#1D5D9B"
          height="1.5em"
          onClick={[Function]}
          stroke="1D5D9B"
          strokeWidth={0}
          viewBox="0 0 24 24"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 11H7.101l.001-.009a4.956 4.956 0 0 1 .752-1.787 5.054 5.054 0 0 1 2.2-1.811c.302-.128.617-.226.938-.291a5.078 5.078 0 0 1 2.018 0 4.978 4.978 0 0 1 2.525 1.361l1.416-1.412a7.036 7.036 0 0 0-2.224-1.501 6.921 6.921 0 0 0-1.315-.408 7.079 7.079 0 0 0-2.819 0 6.94 6.94 0 0 0-1.316.409 7.04 7.04 0 0 0-3.08 2.534 6.978 6.978 0 0 0-1.054 2.505c-.028.135-.043.273-.063.41H2l4 4 4-4zm4 2h2.899l-.001.008a4.976 4.976 0 0 1-2.103 3.138 4.943 4.943 0 0 1-1.787.752 5.073 5.073 0 0 1-2.017 0 4.956 4.956 0 0 1-1.787-.752 5.072 5.072 0 0 1-.74-.61L7.05 16.95a7.032 7.032 0 0 0 2.225 1.5c.424.18.867.317 1.315.408a7.07 7.07 0 0 0 2.818 0 7.031 7.031 0 0 0 4.395-2.945 6.974 6.974 0 0 0 1.053-2.503c.027-.135.043-.273.063-.41H22l-4-4-4 4z"
          />
        </svg>
      </div>
      <div
        className="bg-gray-300 cursor-pointer flex flex-row py-4 px-2 justify-center items-center border-b-2"
        onClick={[Function]}
      >
        <div
          className="w-1/4"
        >
          <div
            className="relative inline-flex items-center justify-center w-10 h-10 overflow-hidden bg-gray-100 rounded-full"
          >
            <span
              className="font-medium text-gray-600"
            >
              P
            </span>
          </div>
        </div>
        <div
          className="w-full"
        >
          <div
            className="text-lg font-semibold"
          >
            public
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full h-[92vh] px-5 flex flex-col justify-between"
    >
      <div
        className="flex flex-col mt-5 overflow-y-auto"
      >
        <div />
      </div>
      <div
        className="py-5"
      >
        <form
          onSubmit={[Function]}
        >
          <label
            className="sr-only"
            htmlFor="chat"
          >
            Your message
          </label>
          <div
            className="flex items-center px-3 py-2 rounded-lg bg-gray-50"
          >
            <textarea
              className="block mx-4 p-2.5 w-full text-sm text-gray-900 bg-white rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500"
              name="input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Your message..."
              rows={1}
            />
            <button
              className="inline-flex justify-center p-2 text-blue-600 rounded-full cursor-pointer hover:bg-blue-100"
              type="submit"
            >
              <svg
                aria-hidden="true"
                className="w-5 h-5 rotate-90"
                fill="currentColor"
                viewBox="0 0 18 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m17.914 18.594-8-18a1 1 0 0 0-1.828 0l-8 18a1 1 0 0 0 1.157 1.376L8 18.281V9a1 1 0 0 1 2 0v9.281l6.758 1.689a1 1 0 0 0 1.156-1.376Z"
                />
              </svg>
              <span
                className="sr-only"
              >
                Send message
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Inbox /> > Inbox renders properly 1`] = `
<div
  className="container mx-auto shadow-lg rounded-lg"
>
  <div
    className="flex flex-row justify-between"
  >
    <div
      className="flex flex-col w-2/5 border-r-2 overflow-y-auto"
    >
      <div
        className="flex justify-end cursor-pointer"
      >
        <span
          className="mr-1"
        >
          Refresh Users
        </span>
        <svg
          className=""
          fill="#1D5D9B"
          height="1.5em"
          onClick={[Function]}
          stroke="1D5D9B"
          strokeWidth={0}
          viewBox="0 0 24 24"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 11H7.101l.001-.009a4.956 4.956 0 0 1 .752-1.787 5.054 5.054 0 0 1 2.2-1.811c.302-.128.617-.226.938-.291a5.078 5.078 0 0 1 2.018 0 4.978 4.978 0 0 1 2.525 1.361l1.416-1.412a7.036 7.036 0 0 0-2.224-1.501 6.921 6.921 0 0 0-1.315-.408 7.079 7.079 0 0 0-2.819 0 6.94 6.94 0 0 0-1.316.409 7.04 7.04 0 0 0-3.08 2.534 6.978 6.978 0 0 0-1.054 2.505c-.028.135-.043.273-.063.41H2l4 4 4-4zm4 2h2.899l-.001.008a4.976 4.976 0 0 1-2.103 3.138 4.943 4.943 0 0 1-1.787.752 5.073 5.073 0 0 1-2.017 0 4.956 4.956 0 0 1-1.787-.752 5.072 5.072 0 0 1-.74-.61L7.05 16.95a7.032 7.032 0 0 0 2.225 1.5c.424.18.867.317 1.315.408a7.07 7.07 0 0 0 2.818 0 7.031 7.031 0 0 0 4.395-2.945 6.974 6.974 0 0 0 1.053-2.503c.027-.135.043-.273.063-.41H22l-4-4-4 4z"
          />
        </svg>
      </div>
      <div
        className="bg-gray-300 cursor-pointer flex flex-row py-4 px-2 justify-center items-center border-b-2"
        onClick={[Function]}
      >
        <div
          className="w-1/4"
        >
          <div
            className="relative inline-flex items-center justify-center w-10 h-10 overflow-hidden bg-gray-100 rounded-full"
          >
            <span
              className="font-medium text-gray-600"
            >
              P
            </span>
          </div>
        </div>
        <div
          className="w-full"
        >
          <div
            className="text-lg font-semibold"
          >
            public
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full h-[92vh] px-5 flex flex-col justify-between"
    >
      <div
        className="flex flex-col mt-5 overflow-y-auto"
      >
        <div />
      </div>
      <div
        className="py-5"
      >
        <form
          onSubmit={[Function]}
        >
          <label
            className="sr-only"
            htmlFor="chat"
          >
            Your message
          </label>
          <div
            className="flex items-center px-3 py-2 rounded-lg bg-gray-50"
          >
            <textarea
              className="block mx-4 p-2.5 w-full text-sm text-gray-900 bg-white rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500"
              name="input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Your message..."
              rows={1}
            />
            <button
              className="inline-flex justify-center p-2 text-blue-600 rounded-full cursor-pointer hover:bg-blue-100"
              type="submit"
            >
              <svg
                aria-hidden="true"
                className="w-5 h-5 rotate-90"
                fill="currentColor"
                viewBox="0 0 18 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m17.914 18.594-8-18a1 1 0 0 0-1.828 0l-8 18a1 1 0 0 0 1.157 1.376L8 18.281V9a1 1 0 0 1 2 0v9.281l6.758 1.689a1 1 0 0 0 1.156-1.376Z"
                />
              </svg>
              <span
                className="sr-only"
              >
                Send message
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
